### DISCUSSION TITLE
Error: bad X server connection; trying to run & visualize w/ peacock on Windows 10 WSL

### DISCUSSION THREAD

**Discussion by ghost** [idaholab/moose#20067] [Q&A Getting Started] [answered]
I'm trying to install MOOSE onto my Windows 10 machine via WSL 2. I have Ubuntu installed, and I followed along the directions found on https://mooseframework.inl.gov/getting_started/installation/windows10.html and https://mooseframework.inl.gov/getting_started/installation/conda.html . I installed VcXsrv with the default settings in the installer, and ran it with variations of native opengl & access control disabled & enabled before opening WSL. I am trying to run & visualize step1.i in the darcy_thermo_mech tutorial with the command: " ~/projects/moose/python/peacock/peacock -i step1.i", but I get the following error:

2022-01-17 18:09:52.911 (   0.084s) [        F8154740]vtkXOpenGLRenderWindow.:448    ERR| vtkXOpenGLRenderWindow (0x55d0e77fb1e0): bad X server connection. DISPLAY=localhost:0. Aborting.

I'm assuming there are additional installations/configurations to be made beyond what is given in the two links above. Would some be willing to help me out? I apologize if this is an error of overlooking something very obvious.

**Comment by GiudGiud** [comment, marked as answer]
Hello

Peacock isnt working very well with WSL. Even if we figure this one out, it'll show artifacts during rendering.
Did some googling and it could be a firewall issue. 
Others are saying you need to change the DISPLAY env variables https://superuser.com/questions/1476086/error-cant-open-display-0

Guillaume

**Reply by ghost** [reply]
Would using Paraview be a suitable alternative for visualization in WSL?

**Reply by GiudGiud** [reply]
Yes. Though what I did when I was working in WSL was to install paraview on windows and read the files on the windows side.

**Reply by ghost** [reply]
What's the easiest way you've found to transfer files in the WSL 'virtual directories' to the Windows directories?

**Reply by GiudGiud** [reply]
I think I could just double click on the file navigator to open these files.
I was using mobaxterm

**Reply by ghost** [reply]
> Yes. Though what I did when I was working in WSL was to install paraview on windows and read the files on the windows side.

Works like a charm. Thank you!
